doc/compiler/control/CompilationControl.md

```diff
@@ -0,0 +1,304 @@
+1 bit.
+
+When the method is invoked, the VM decrements the count in the
+`extra` field. When the count hits zero, the VM notifies the
+JIT to schedule a compilation. Once the compilation succeeds,
+the `extra` field is updated with the address of the Start PC of
+the compiled body. Because the `extra` field's LSB is used to test
+whether the method is interpreted or compiled, the Start PC of a
+compiled method body is always at least 2-byte aligned.
+
+It is also worth noting that
+[Async Checkpoint Sampling](#async-checkpoint-sampling)
+can also result in artificially decrementing the invocation count
+to cause the method to get compiled sooner if it is determined
+that the method is important for performance.
+
+### Recompilation
+
+Recompilations are triggered in a multitude of ways; see the
+[Recompilation doc](../runtime/Recompilation.md) for more details.
+
+### Dynamic Loop Transfer
+
+Dynamic Loop Transfer (DLT) is a feature in OpenJ9 that allows
+a thread executing in the middle of an interpreted method to
+switch to the middle of a compiled method body and continue
+execution. This feature is used for methods that spend a lot of
+time in the interpreter but are not invoked many times; this
+happens when executing a long running loop.
+
+The JVM maintains an DLT buffer per Java Thread. Using
+[Async Checkpoint Sampling](#async-checkpoint-sampling),
+if the JIT determines that a method is a long running
+interpreted method that would benefit from DLT, the method is
+queued for both a DLT and regular compilation. On a succesful
+DLT compilation, the VM transitions the thread to execute the
+compiled version of the method.
+
+It is worth noting that a DLT compiled method body can only be
+transitioned to from a specific Bytecode PC. It is for this
+reason that a regular compilation is also requested along with
+the DLT compilation; the former ensures that new invocations
+of the method do not have to run interpreted until it reaches
+the Bytecode PC that permits the transition to the DLT compiled
+method body.
+
+
+## Compilation Queue
+
+The Compilation Queue is a weighted queue of compilation
+entries. The weights represent the priority of a compilation;
+in general, the cheaper a compile is expected to be, the
+higher its priority. The queue is synchronized using the
+[Compilation Monitor](#compilation-monitor).
+
+### Low Priority Queue
+
+The JIT also maintains a Low Priority Queue (LPQ). This is
+used to compile methods that are not immediately needed for
+performance, but would benefit the application in the long
+run. Entries from the LPQ are only picked up by Compilation
+Threads when the main Compilation Queue is empty.
+
+
+## Compilation Threads
+
+Compilations are performed on dedicated threads known as Compilation
+Threads. These threads are created at JVM startup, and are
+suspended until there is work to be done. The exact number of
+threads created depends on the resources available on the machine,
```

Comment:
on the resources available on the machine   --> on the CPU resources available to the JVM

Reply:
The number of compilation threads created does not depend on the size of the compilation queue. The number of active compilation threads depends on the size of the compilation queue.